### PR TITLE
feat(misk-security): Add @Csp annotation to WebActions.

### PR DESCRIPTION
**Pull request by kerrnel-cash**
This adds an @Csp annotation so that developers can specify Content-Security-Policy for their endpoints.

This was co-authored by frojas@squareup.com

### CODE REVIEW

**Review by frojasg — APPROVED** [review]
LGTM

File: misk/src/main/kotlin/misk/security/csp/Csp.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+//
+// Production CSPs can be very long so a future contribution should add the ability to define
+// CSP into YAML configs and point the @Csp annotation to the config value.
+//
+// This annotation is currently dumb, it adds the rules passed to a Content-Security-Policy
+// header directive, with no inspection or validation. A future change could parse the policy.
+//
+// Developers using this should check out the various browser extensions to verify the actual
+// CSP on their webpages.
+annotation class Csp(val rules: Array<String>)
```
Comment (adrw) [inline misk/src/main/kotlin/misk/security/csp/Csp.kt:15]: I'd lean towards CSP being written out to reduce use of acronyms in code

Comment (kerrnel-cash) [reply misk/src/main/kotlin/misk/security/csp/Csp.kt:15]: I can land a follow up to make it @ContentSecurityPolicy. Francisco, what do you think?

File: misk/src/main/kotlin/misk/security/csp/Csp.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package misk.security.csp
+
+// This annotation allows misk endpoints to define their Content-Security-Policy directive.
```
Comment (yoavamit) [inline misk/src/main/kotlin/misk/security/csp/Csp.kt:3]: nit: change the comment type to proper javadoc format (replace `//` with `/** */`)

File: misk/src/main/kotlin/misk/security/csp/CspInterceptor.kt [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package misk.security.csp
+
+import misk.Action
+import misk.web.NetworkChain
+import misk.web.NetworkInterceptor
+import javax.inject.Inject
+import kotlin.reflect.full.findAnnotation
+
+class CspInterceptor(val rules: List<String>) : NetworkInterceptor {
```
Comment (yoavamit) [inline misk/src/main/kotlin/misk/security/csp/CspInterceptor.kt:9]: missing doc on a public class

File: misk/src/main/kotlin/misk/security/csp/CspInterceptor.kt [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  override fun intercept(chain: NetworkChain) {
+    chain.httpCall.setResponseHeader("Content-Security-Policy", rules.joinToString(separator = "; ", postfix = ";"))
```
Comment (yoavamit) [inline misk/src/main/kotlin/misk/security/csp/CspInterceptor.kt:11]: Thought for a follow up PR - should we also validate the syntax and grammar of the policy?

File: misk/src/main/kotlin/misk/security/csp/Csp.kt (hunk 1 above)
Comment (kerrnel-cash) [reply misk/src/main/kotlin/misk/security/csp/Csp.kt:15]: Done, follow up PR coming.

File: misk/src/main/kotlin/misk/security/csp/CspInterceptor.kt (hunk 3 above)
Comment (kerrnel-cash) [reply misk/src/main/kotlin/misk/security/csp/CspInterceptor.kt:9]: Done, see follow up PR.

File: misk/src/main/kotlin/misk/security/csp/CspInterceptor.kt (hunk 4 above)
Comment (kerrnel-cash) [reply misk/src/main/kotlin/misk/security/csp/CspInterceptor.kt:11]: Yes, but not yet.

**Pull request merged**